### PR TITLE
[KED-1863] Configure base image

### PR DESCRIPTION
**Pull request by DmitriiDeriabinQB**
## Motivation and Context
Configure the base image through the CLI option.

**Note:** Please note that I also changed the default base image from Debian stretch (`9.*`) to Debian buster (`10.*`) since [there is no](https://hub.docker.com/_/python) official stretch image for Python 3.8. Also, [stretch security updates have been discontinued as of July 6th, 2020](https://www.debian.org/releases/stretch/).

## How has this been tested?
e2e tests

## Checklist

- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [x] Assigned myself to the PR
- [x] Added tests to cover my changes
- [x] Added new entries to the `RELEASE.md` file

## Notice

- [x] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

- I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
- I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorised to submit this contribution on behalf of the original creator(s) or their licensees.
- I certify that the use of this contribution as authorised by the Apache 2.0 license does not violate the intellectual property rights of anyone else.

### CODE REVIEW

File: features/docker.feature [hunk 1]
```diff
@@ -50,8 +50,25 @@ Feature: Docker commands in new projects
     Then I should get a successful exit code
     And A new docker image for test project should be created
 
+  Scenario: Execute docker build target with custom base image
+    Given I have removed old docker image of test project
```
Comment (lorenabalan) [inline features/docker.feature:54]: Can we add this step in `background` or does it mess with the last few scenarios that don't have it?

File: README.md [hunk 2]
```diff
@@ -48,9 +48,12 @@ Behind the scenes Kedro does the following:
...
+
 > *Note:* By default, `kedro docker build` creates an image without Spark and Hadoop.
 
-By default, the project Docker image will be tagged as `<project-root-dir>:latest`, where `<project-root-dir>` is the name of the project root directory. To change the tag, you can add the `--image` command line option, for example: `kedro docker build --image my-project-tag`.
+> *Note:* By default, when calling `kedro docker build` image is built with `python:VERSION-buster` image, where VERSION is Python (major + minor) version from the current environment. By specifying `--base-image` option, different base image can be used. For example `kedro docker build --base-image="python:3.8-buster"`.
+
+The project Docker image will automatically be tagged as `<project-root-dir>:latest`, where `<project-root-dir>` is the name of the project root directory. To change the tag, you can add the `--image` command line option, for example: `kedro docker build --image my-project-tag`.
```
Comment (andrii-ivaniuk) [inline README.md:56]: Why `--image` is used to add a tag? Why not `--tag`?

Comment (DmitriiDeriabinQB) [reply README.md:56]: Image is uniquely described by its config sha256 hash, however people don't usually use those and use names instead. Image name has a specific format: `<name>:<tag>`. If tag is omitted, then `latest` is assumed. When "tagging" the image, one should use the whole name, not just "tag", hence the `--image` option. To be perfectly precise, it should be something like `--image-name`.
I agree that this is inconsistent with `docker build` command, which uses `--tag` option, but I think `--image` is also okay.

File: features/docker.feature (hunk 1 above)
Comment (DmitriiDeriabinQB) [reply features/docker.feature]: A very good point, thanks! 👍

**Pull request merged**